scripts/cms5/merge-ups.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+// run 3 days offset from when a weekly CI run is, otherwise run on a random day of the week
+$runOnDay = predictable_random_int(6);
+// run at the same hour as the weekly CI run, otherwise run at a random hour of the day
+$runOnHour = predictable_random_int(23);
+// run at the same minute as the weekly CI run, otherwise run at a random minute of the hour
+// rounded to 5 minutes
+$runOnMinute = predictable_random_int(11) * 5;
+
+if (check_file_exists('.github/workflows/dispatch-ci.yml')) {
+    $ci = read_file('.github/workflows/dispatch-ci.yml');
+    preg_match("#- cron: '(.+?) (.+?) (.+?) (.+?) (.+?)'#", $ci, $matches);
+    [$_, $minute, $hour, $day, $month, $dayOfWeek] = $matches;
+    if ($dayOfWeek !== '*') {
+        $days = explode(',', $dayOfWeek);
+        $runOnDay = ($days[count($days) - 1] + 3) % 7;
+    }
+    if ($hour !== '*') {
+        $hours = explode(',', $hour);
+        $runOnHour = $hours[0];
+    }
+    if ($minute !== '*') {
+        $runOnMinute = $minute;
+    }
```

Comment:
Don't think it's necessary to run at the same hour and minute - but I don't feel strongly enough to explicitly ask for it to be removed if you think it makes sense to have there.

Reply:
Leave as is